jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -72,57 +73,70 @@ public static void configureMode(Server server, String mode)
     public enum Mode
     {
         GENERATE,  // Generate quickstart-web.xml and then stop
-        AUTO,      // use or generate depending on the existance of quickstart-web.xml
+        AUTO,      // use or generate depending on the existence of quickstart-web.xml
         QUICKSTART // Use quickstart-web.xml
     }
 
-    private Mode _mode = Mode.AUTO;
-    private boolean _quickStart;
-
     public QuickStartConfiguration()
     {
         super(true);
         addDependencies(WebInfConfiguration.class);
         addDependents(WebXmlConfiguration.class);
     }
 
+    private Mode getModeForContext(WebAppContext context)
+    {
+        Mode mode = (Mode)context.getAttribute(MODE);
+        if (mode == null)
+            return DEFAULT_MODE;
+        return mode;
+    }
+
     @Override
     public void preConfigure(WebAppContext context) throws Exception
     {
-        //check that webapp is suitable for quick start - it is not a packed war
+        // check that webapp is suitable for quick start - it is not a packed war
         String war = context.getWar();
-        if (war == null || war.length() <= 0 || !context.getBaseResource().isDirectory())
-            throw new IllegalStateException("Bad Quickstart location");
+        if (StringUtil.isBlank(war) || !context.getBaseResource().isDirectory())
+            throw new IllegalStateException("Invalid Quickstart location");
 
-        //look for quickstart-web.xml in WEB-INF of webapp
+        // look for quickstart-web.xml in WEB-INF of webapp
         Resource quickStartWebXml = getQuickStartWebXml(context);
-        LOG.debug("quickStartWebXml={} exists={}", quickStartWebXml, quickStartWebXml.exists());
+        if (quickStartWebXml == null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("No quickStartWebXml present on {}", context);
+            return;
+        }
 
-        //Get the mode
-        Mode mode = (Mode)context.getAttribute(MODE);
-        if (mode != null)
-            _mode = mode;
-        
-        _quickStart = false;
-        
-        switch (_mode)
+        // Get the mode
+        Mode mode = getModeForContext(context);
+
+        if (LOG.isDebugEnabled())
+            LOG.debug("mode={} quickStartWebXml={} exists={} for {}",
+                mode,
+                quickStartWebXml,
+                quickStartWebXml.exists(),
+                context);
+
+        switch (mode)
         {
             case GENERATE:
             {
                 if (quickStartWebXml.exists())
-                    LOG.info("Regenerating {}", quickStartWebXml);
+                    LOG.info("Regenerating {} for {}", quickStartWebXml, context);
                 else
-                    LOG.info("Generating {}", quickStartWebXml);
-                
-                super.preConfigure(context);
```

Comment:
All configs call `super.preConfigure()`, so safest to leave it in.

Reply:
Incorrect assumption.
See https://github.com/jetty/jetty.project/pull/11261#discussion_r1452613152